app/cluster/integration_test.go

```diff
@@ -285,10 +285,16 @@ func TestDynamicClusterManager(t *testing.T) {
 	require.Eventually(t, func() bool {
 		return dCM.Mode() == servermode.NormalMode
 	}, time.Second, time.Millisecond)
+
+	// Letting the router start function to return
+	// else stop would be waiting forever on the wait-group
+	time.Sleep(10 * time.Second)
```

Comment:
The test is not passing anyway in my machine. I'll debug this a bit and let you know.

Reply:
@cisse21 I pushed a fix for the router manager [here](https://github.com/rudderlabs/rudder-server/pull/2235/commits/a8a24673e189d6598e76e84ab0acccec39619cc6). Now the test passes with no sleep nor delays in my machine. Let me know what you think. Now I'm going to create a ticket to tackle the outstanding router issues.

Reply:
Additional fix pushed [here](https://github.com/rudderlabs/rudder-server/pull/2235/commits/512d31c13c624899cac0237b156475e146942c35).

Reply:
Ticket [here](https://www.notion.so/rudderstacks/BackendConfig-is-initialized-when-credentials-are-bad-e0c2d684739948afa5b99a2c1c48afae).